kayenta-sql/src/main/java/com/netflix/kayenta/sql/migration/StorageDataMigrator.java

```diff
@@ -0,0 +1,110 @@
+import java.util.concurrent.ExecutorService;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.scheduling.annotation.Scheduled;
+
+@Slf4j
+@RequiredArgsConstructor
+public class StorageDataMigrator {
+
+  private final SqlProperties sqlProperties;
+  private final StorageService sourceStorageService;
+  private final StorageService targetStorageService;
+  private final ExecutorService executorService;
+
+  private void migrate(ObjectType objectType) {
+    log.info("Migrating {}", objectType);
+
+    var sourceAccountName = sqlProperties.getMigration().getSourceAccountName();
+    var targetAccountName = sqlProperties.getMigration().getTargetAccountName();
+
+    var sourceObjectKeys =
+        sourceStorageService.listObjectKeys(sourceAccountName, objectType).stream()
+            .map(sourceObjectKey -> (String) sourceObjectKey.get("id"))
+            .collect(Collectors.toList());
+
+    var targetObjectKeys =
+        targetStorageService.listObjectKeys(targetAccountName, objectType).stream()
+            .map(sourceObjectKey -> (String) sourceObjectKey.get("id"))
+            .collect(Collectors.toList());
+
+    var objectKeysToMigrate =
+        sourceObjectKeys.stream()
+            .filter(
+                sourceObjectKey ->
+                    targetObjectKeys.stream()
+                        .filter(targetObjectKey -> Objects.equals(targetObjectKey, sourceObjectKey))
+                        .findFirst()
+                        .map(
+                            object -> {
+                              log.warn(
+                                  "Object for objectType: {}, key: {} already exists",
+                                  objectType,
+                                  sourceObjectKey);
+                              return object;
+                            })
+                        .isEmpty())
+            .collect(Collectors.toList());
+
+    if (objectKeysToMigrate.isEmpty()) {
+      log.info(
+          "No objects to migrate for objectType: {}, sourceObjectCount: {}, targetObjectCount: {}",
+          objectType,
+          sourceObjectKeys.size(),
+          targetObjectKeys.size());
+      return;
+    }
+
+    for (var objectKey : objectKeysToMigrate) {
+      executorService.submit(
+          () -> {
+            try {
+              var object =
+                  sourceStorageService.loadObject(sourceAccountName, objectType, objectKey);
+              targetStorageService.storeObject(targetAccountName, objectType, objectKey, object);
+            } catch (Exception e) {
+              log.error(
+                  "Unable to migrate objectType: {}, objectKey: {}", objectType, objectKey, e);
+            }
+          });
+    }
+  }
+
+  @Scheduled(fixedDelay = 60000)
+  public void migrate() {
+    if (!sqlProperties.getMigration().isEnabled()) {
+      log.info("Migration is disabled");
```

Comment:
Given this class should ONLY load if migration is enabled... isn't this somewhat pointless?